cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -374,13 +403,23 @@ func (c *CStorVolumeReplicaController) cVRAddEventHandler(
 		quorum = false
 	}
 
+	//TODO: Follow best practice while refactor reconciliation logic
+
+	// Following sinppet will do following things
+	// 1. Verify whether volume replica needs to create or not? If yes proceed
+	//    further.
+	// 2. If replicaID is empty and if it is new volume generate replicaID.
+	// 3. Trigger ZFS create command on success get the status from ZFS and
+	//    update it. If `ZFS command` fails then return with same status phase
+	//    which is currently holding by CVR.
 	// IsEmptyStatus is to check if initial status of cVR object is empty.
 	if IsEmptyStatus(cVR) || IsInitStatus(cVR) || IsRecreateStatus(cVR) {
-		// We should generate replicaID for new volume only
-		if IsEmptyStatus(cVR) || IsInitStatus(cVR) {
+		// We should generate replicaID for new volumes only if it doesn't has
+		// replica ID.
+		if isEmptyReplicaID(cVR) && (IsEmptyStatus(cVR) || IsInitStatus(cVR)) {
 			if err := volumereplica.GenerateReplicaID(cVR); err != nil {
```

Comment:
check if this snippet of code can be refactored, as we are having a same conditions check above and below.

Reply:
Not sure how to avoid these checks with current code